### ISSUE TITLE
npm module does not install dependencies globally

### ISSUE THREAD

**Issue by majuric** [ansible/ansible-modules-extras#1675]
Issue Type:

Bug Report

Ansible Version:

2.0.0.2

Environment:

Host
Linux Mint 17
Python 2.7.6

Server
Ubuntu server 14.04

Summary:

Ansible npm task is performed successfully ( or so it would seem ) but the packages don't get installed globally

Steps To Reproduce:

```
- name: Install required node packages
  npm:
    name: "{{ item.name }}"
    global: yes
    state: latest
  with_items:
  - name: grunt-cli
  - name: bower
```

Expected Results:

The npm packages to be available globally 

Actual Results:

Packages do not get installed, even though the response from ansible is that the task was successful 

> changed: [192.168.1.90] => (item={u'name': u'grunt-cli'}) => {"changed": true, "invocation": {"module_args": {"executable": null, "global": true, "ignore_scripts": false, "name": "grunt-cli", "path": null, "production": false, "registry": null, "state": "latest", "version": null}, "module_name": "npm"}, "item": {"name": "grunt-cli"}}
> changed: [192.168.1.90] => (item={u'name': u'bower'}) => {"changed": true, "invocation": {"module_args": {"executable": null, "global": true, "ignore_scripts": false, "name": "bower", "path": null, "production": false, "registry": null, "state": "latest", "version": null}, "module_name": "npm"}, "item": {"name": "bower"}}

when I ssh into the machine, and try running grunt or bower from the command line I get an error saying the packaged are not available.

Installing the packages manually works as expected.

**Comment by sairez** [comment]
seeing the same in 

```
~$ ansible --version
ansible 2.0.0.2
  config file = 
  configured module search path = Default w/o overrides
```

**Comment by fubarhouse** [comment]
```
- name: "NPM | Ensure installed and updated"
  become: yes
  become_user: "{{ ansible_ssh_user }}"
  npm:
    name: "{{ item.name }}"
    executable: "{{ fubarhouse_npm.npm_install_exec }}"
    path: "/home/{{ ansible_ssh_user }}/.npm/"
    state: latest
    global: yes
  with_items:
    - "{{ fubarhouse_npm.packages }}"
```

Same problem, but I can't find the packages anywhere on the system after the provisioning is complete.
Although, installing them manually seems to work fine.

edit: It's worth noting that I'm experiencing this same problem with the gem module, and I've never actually had a working copy of these modules work. I'm using ubuntu 12.04.

**Comment by jm-welch** [comment]
Same issue here.

Ubuntu 14.04 server

```
$ ansible --version
ansible 2.0.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

``` yaml
- name: Install yeoman (yo)
  npm:
    name: yo
    state: latest
    global: True
```

After I run the play, 

``` shell
$ yo
yo: command not found

$ ls /usr/lib/node_modules
npm
```

I checked the prefix (default, I haven't touched it), and it's set to `/usr`. If I install manually from npm in the shell (`npm install yo -g`), everything works.

This does _not_ happen on my CentOS 7 box, only (so far) on Ubuntu. On my CentOS box, modules installed with `global: yes` (same playbook, actually) get placed into `/usr/lib/node_modules`, as expected.

``` shell
#CentOS
$ sudo npm get prefix -g
/

#Ubuntu
$ sudo npm get prefix -g
/usr
```

Could the prefix be the issue?

**Comment by omikron-admin** [comment]
Hi,
the same problem on our ubuntu 14.04, the modules are not being installed

**Comment by Cbeck527** [comment]
As addressed in #1375, this is actually something broken when you use `state=latest`. Using `state=present` will install the module globally as expected. 

If you really need the latest, then split it up into two tasks, one ensuring that the module is present and the second ensuring that it at the latest version! Here's hoping this is fixed soon 👍 

(workaround credit: @geerlingguy 🙌)

**Comment by ansibot** [comment]
@chrishoffman, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@shane-walker, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by pokidovea** [comment]
Ping! It's still reproducible on ansible v2.1.1.0

**Comment by ansibot** [comment]
@shane-walker, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@shane-walker, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by tommygnr** [comment]
Can this be closed in light of https://github.com/ansible/ansible-modules-extras/issues/1375 ?

**Comment by ansibot** [comment]
@shane-walker, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@shane-walker, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
This issue was migrated to https://github.com/ansible/ansible/issues/29271
